enterprise/coderd/licenses_test.go

```diff
@@ -11,12 +14,291 @@ import (
 	"github.com/stretchr/testify/require"
 	"golang.org/x/xerrors"
 
+	"github.com/coder/coder/v2/coderd/coderdtest"
 	"github.com/coder/coder/v2/codersdk"
 	"github.com/coder/coder/v2/enterprise/coderd/coderdenttest"
 	"github.com/coder/coder/v2/enterprise/coderd/license"
+	"github.com/coder/coder/v2/enterprise/coderd/usage/tallymansdk"
 	"github.com/coder/coder/v2/testutil"
 )
 
+func TestPostUsageEmbeddableDashboard(t *testing.T) {
+	t.Parallel()
+
+	t.Run("NoLicense", func(t *testing.T) {
+		t.Parallel()
+
+		client, _ := coderdenttest.New(t, &coderdenttest.Options{
```

Comment:
Let's not make the problem testing `coderd` worse by spinning up a whole API server & database to test one handler.  Can we organize this like the agentapiserver so that it is separable from Coderd and we can test with standard `httptest` tools?

Reply:
It's not every other endpoint, e.g. agentapiserver